apps/files/models.py

```diff
@@ -131,10 +131,15 @@ def get_mirror(self, addon, attachment=False):
         return posixpath.join(*map(smart_str, [host, addon.id, self.filename]))
 
     def get_url_path(self, src):
+        return self._make_download_url('downloads.file', src)
+
+    def get_signed_url_path(self, src):
+        return self._make_download_url('signing.file', src)
+
+    def _make_download_url(self, view_name, src):
         from amo.helpers import urlparams, absolutify
-        url = os.path.join(reverse('downloads.file', args=[self.id]),
+        url = os.path.join(reverse(view_name, args=[self.pk]),
```

Comment:
I lied about lying `urlparse.urljoin` would work you just get different results based on the trailing slash.

```py
In [30]: urlparse.urljoin('/file/1000/', 'filename.xpi')
Out[30]: '/file/1000/filename.xpi'
In [31]: urlparse.urljoin('/file/1000', 'filename.xpi')
Out[31]: '/file/filename.xpi'